### PR TITLE
Tanvir dev

### PR DESCRIPTION
**Pull request by Tanvir-Rahman123**
-Video API fully functioning on home page and creator and profile pages with pagination
- Sort feature added to creator and profile pages

### CODE REVIEW

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 1]
```diff
@@ -37,12 +37,79 @@ import Tab from "@material-ui/core/Tab";
...
+            selected={option === sort}
+            onClick={handleClose}
+          >
+            {option}
+          </MenuItem>
+        ))}
+      </Menu>
+    </div>
+  );
+}
```
Comment (R313mon) [inline emuu-client/src/components/CreatorsPage/Feeds.js:100]: Good use of menu tab to sort videos.

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 2]
```diff
@@ -37,12 +37,79 @@ import Tab from "@material-ui/core/Tab";
...
 
 function Feeds({ setVideo, setUserProfile }) {
   const [recentVideos, setRecentVideos] = useState([]);
+  const [topVideos, setTopVideos] = useState([]);
+  const [pages, setPages] = useState(undefined);
+  const [page, setPage] = useState(1);
```
Comment (R313mon) [inline emuu-client/src/components/CreatorsPage/Feeds.js:106]: Pages and Page are too similar variables, try making it a little more descriptive to allow code to be read easier.

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 3]
```diff
@@ -55,39 +122,34 @@ function Feeds({ setVideo, setUserProfile }) {
...
+    await axios
+      .post(
+        "http://localhost:8080/auth/video",
+        JSON.stringify({ ...disAndPage })
+      )
+      .then(function (response) {
+        console.log(response);
+      });
+    try {
+      const response = await axios.get("http://localhost:8080/auth/video");
```
Comment (R313mon) [inline emuu-client/src/components/CreatorsPage/Feeds.js:139]: good use of axios post and get to send and retrieve date from the video api.

File: emuu-client/src/components/CreatorsPage/UserInfo.js [hunk 4]
```diff
@@ -25,7 +24,6 @@ function UserInfo() {
   });
 
   return (
-
     <div className="Main-UserInfo-Container">
       <h3 className="about">About</h3>
```
Comment (R313mon) [inline emuu-client/src/components/CreatorsPage/UserInfo.js:29]: Good use of prettier, I see its becoming a more consistent thing all of us are utilizing.

File: emuu-client/src/components/NavbarPostLogin/NavBarNoSearch.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+            <ProfileMenu />
+          </Container>
+        </Navbar>
+      ))}
+    </>
+  );
+}
+
+export default NavBarNoSearch;
```
Comment (R313mon) [inline emuu-client/src/components/NavbarPostLogin/NavBarNoSearch.js:91]: I dont think we are using this anymore, should delete this whole file before merging to main.

File: emuu-client/src/components/UserAuthentication/newloginscreen.js [hunk 6]
```diff
@@ -172,7 +176,7 @@ function Login() {
                 </Button>
                 <Grid container>
                   <Grid item>
-                    <Link className ="registerLink" href="/Register">
+                    <Link className="registerLink" href="/Register">
                       Don't have an account? Sign Up
                     </Link>
                   </Grid>
```
Comment (R313mon) [inline emuu-client/src/components/UserAuthentication/newloginscreen.js:182]: Rename file to just login screen

File: emuu-client/src/components/UserProfile/Feeds.js [hunk 7]
```diff
@@ -174,7 +248,67 @@ function Feeds({ setVideo }) {
...
+                      <div className="videoInfo">
+                        <Typography
+                          variant="body2"
+                          color="text.secondary"
+                          fontWeight="medium"
+                          fontSize="18px"
+                        >
+                          {" "}
+                          {video.Username} &ensp;&ensp;&ensp;&ensp;&ensp;
+                          {video.Likes} Likes &#x2022; {video.Views} Views
```
Comment (R313mon) [inline emuu-client/src/components/UserProfile/Feeds.js:302]: Fix this after merge, I updated it so likes and username were on different lines but it seems to be overwritten.

File: emuu-client/src/components/UserProfile/Feeds.js [hunk 8]
```diff
@@ -187,10 +321,7 @@ function Feeds({ setVideo }) {
                     <CardContent>
                       <CardHeader
                         avatar={
-                          <Avatar
-                            sx={{ width: 60, height: 60 }}
-
-                          ></Avatar>
+                          <Avatar sx={{ width: 60, height: 60 }}></Avatar>
```
Comment (R313mon) [inline emuu-client/src/components/UserProfile/Feeds.js]: Note: Make this smaller (clients request)

File: emuu-client/src/components/home/Home.js [hunk 9]
```diff
@@ -34,7 +34,8 @@ import FavoriteIcon from "@mui/icons-material/Favorite";
 import ShareIcon from "@mui/icons-material/Share";
```
Comment (R313mon) [inline emuu-client/src/components/home/Home.js:34]: Some of these imports are never used.

Comment (R313mon) [reply emuu-client/src/components/home/Home.js:34]: Clean up before merge with main

File: emuu-server/users/upload.go [hunk 10]
```diff
@@ -1,90 +1,132 @@
...
+}
+func (u *UploadInfo) setTags(tags string) {
+	u.Game_tags = tags
+}
+func (u *UploadInfo) setvidUrl(url string) {
+	u.Video_url = url
+}
+func (u *UploadInfo) setthumbUrl(url string) {
+	u.Thumbnail_url = url
+}
```
Comment (R313mon) [inline emuu-server/users/upload.go:69]: Great job making upload object orientated.

File: emuu-server/video/video.go [hunk 11]
```diff
@@ -85,24 +91,42 @@ func SetVideos(c *gin.Context) {
...
+				recentArr = recentArr[(PageNum-1)*6 : len(recentArr)]
+
+			}
+		}
+		if len(mostViewedArr) > 6 {
+			if len(mostViewedArr) > PageNum*6 {
+				mostViewedArr = mostViewedArr[(PageNum-1)*6 : (PageNum)*6]
 
+			} else {
+				mostViewedArr = mostViewedArr[(PageNum-1)*6 : len(mostViewedArr)]
```
Comment (R313mon) [inline emuu-server/video/video.go:118]: This array should contain 8 not 6 videos.

File: emuu-server/video/video.go [hunk 12]
```diff
@@ -131,20 +155,43 @@ func SetVideos(c *gin.Context) {
...
+			Pages        int
 		}{
-			MostViewed:   mostViewedArr[0:7],
+
+			MostViewed:   mostViewedArr,
 			RecentUpload: recentArr,
+			Pages:        pageAmount,
 		}
 
 		c.JSON(http.StatusOK, gin.H{"message": response})
```
Comment (R313mon) [inline emuu-server/video/video.go:197]: Please comment code to allow it to become easier to follow. Overall good logic on this API.

**Review by R313mon — COMMENTED** [review]
Overall good work, just need to remove unused import statements after all of us merge with PR3-Dev to prevent merge conflicts. Definetly need to comment more on codes. Minor tweaks to be made after this merge because I do see something have been overwritten.

**Review by R313mon — APPROVED** [review]

File: emuu-client/src/components/CreatorsPage/CreatorsPage.js [hunk 13]
```diff
@@ -66,20 +62,22 @@ function Creator({ setVideo, video}) {
             <div className="user-profile-bg-image">
               <img id="prf-bg-img" src={Banner} alt="" srcSet="" />
             </div>
-            </div>
+          </div>
 
-              <div className="middle-portion">
+          <div className="middle-portion">
```
Comment (Moamin1) [inline emuu-client/src/components/CreatorsPage/CreatorsPage.js]: Why do we have this middle-portion?

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 14]
```diff
@@ -55,39 +122,34 @@ function Feeds({ setVideo, setUserProfile }) {
...
+      displayName: displayName,
+      pageNumber: page.toString(),
+    };
+    await axios
+      .post(
+        "http://localhost:8080/auth/video",
+        JSON.stringify({ ...disAndPage })
+      )
+      .then(function (response) {
+        console.log(response);
```
Comment (Moamin1) [inline emuu-client/src/components/CreatorsPage/Feeds.js:136]: delete this console.log statement

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 15]
```diff
@@ -55,39 +122,34 @@ function Feeds({ setVideo, setUserProfile }) {
...
-          let temp = videos[i];
-          videos[i] = videos[i + 1];
-          videos[i + 1] = temp;
-        }
-      }
+      setTopVideos(response.data.message.MostViewed);
+      setRecentVideos(response.data.message.RecentUpload);
+      setPages(response.data.message.Pages);
+    } catch (error) {
+      console.log(error);
```
Comment (Moamin1) [inline emuu-client/src/components/CreatorsPage/Feeds.js:145]: delete this console.log as well

**Review by Moamin1 — APPROVED** [review]
Great job on getting the videos sorted and displayed from the backend with pagination.

**Pull request merged**